readthedocs/api/v3/tests/test_projects.py

```diff
@@ -181,3 +181,92 @@ def test_import_project_with_extra_fields(self):
+            'privacy_level': 'private',
+            'analytics_code': 'UA-XXXXXX',
+            'show_version_warning': False,
+            'single_version': True,
+        }
+
+        self.client.credentials(HTTP_AUTHORIZATION=f'Token {self.token.key}')
+        response = self.client.put(
+            reverse(
+                'projects-detail',
+                kwargs={
+                    'project_slug': self.project.slug,
+                },
+            ),
+            data,
+        )
+        self.assertEqual(response.status_code, 204)
+
+        self.project.refresh_from_db()
+        self.assertEqual(self.project.name, 'Updated name')
+        self.assertEqual(self.project.slug, 'project')
+        self.assertEqual(self.project.repo, 'https://bitbucket.com/rtfd/updated-repository')
+        self.assertEqual(self.project.repo_type, 'hg')
+        self.assertEqual(self.project.language, 'es')
+        self.assertEqual(self.project.programming_language, 'js')
+        self.assertEqual(self.project.project_url, 'https://updated-homepage.org')
+        self.assertEqual(self.project.default_version, 'stable')
+        self.assertEqual(self.project.default_branch, 'updated-default-branch')
+        self.assertEqual(self.project.privacy_level, 'private')
+        self.assertEqual(self.project.analytics_code, 'UA-XXXXXX')
+        self.assertEqual(self.project.show_version_warning, False)
+        self.assertEqual(self.project.single_version, True)
+
+    def test_partial_update_project(self):
+        data = {
+            'name': 'Updated name',
+            'repository': {
+                'url': 'https://github.com/rtfd/updated-repository',
+            },
+            'default_branch': 'updated-default-branch',
+        }
+
+        self.client.credentials(HTTP_AUTHORIZATION=f'Token {self.token.key}')
+        response = self.client.patch(
+            reverse(
+                'projects-detail',
+                kwargs={
+                    'project_slug': self.project.slug,
+                },
+            ),
+            data,
+        )
+        self.assertEqual(response.status_code, 204)
+
+        self.project.refresh_from_db()
+        self.assertEqual(self.project.name, 'Updated name')
+        self.assertEqual(self.project.slug, 'project')
+        self.assertEqual(self.project.repo, 'https://github.com/rtfd/updated-repository')
+        self.assertNotEqual(self.project.default_version, 'updated-default-branch')
+
+    def test_partial_update_others_project(self):
+        data = {
+            'name': 'Updated name',
+        }
+
+        self.client.credentials(HTTP_AUTHORIZATION=f'Token {self.token.key}')
+        response = self.client.patch(
+            reverse(
+                'projects-detail',
+                kwargs={
+                    'project_slug': self.others_project.slug,
+                },
+            ),
+            data,
+        )
+        self.assertEqual(response.status_code, 403)
```

Comment:
This is returning 204, allowing a user to modify other's people project. There is a bug where we check for permissions that I need to fix before merging this.

Reply:
This is fixed in last commits.